File: pingyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import java.util.List;
import java.util.Map;

import com.pinyougou.pojo.TbBrand;
import entry.PageResult;

/**
 * 品牌接口
 * @author Administrator
 *
 */
public interface BrandService {

	//查询全部品牌数据
	public List<TbBrand> findAll();
	//分页查询品牌数据
	public PageResult findPage(int pageNum,int pageSize);
	//增加品牌
	public void addBrand(TbBrand brand);
	//修改品牌(先查询单个品牌数据)
	public TbBrand findById(Long id);
	//修改
	public void updateBrand(TbBrand brand);
	//批量删除数据
	public void deleteBrand(Long[] ids);
	//按条件查询
	public PageResult findPageSearch(TbBrand brand, int pageNum,int pageSize);

	public List<Map> selectOptionList();
}

File: pingyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service('brandService',function($http){
    //查询全部
    this.findAll=function(){
        return $http.get('/brand/findAll');
    }
    //分页查询全部
    this.findPage=function (page,rows) {
        return $http.get('/brand/findPage?page='+page+'&rows='+rows);
    }
    //查询单个
    this.findById=function (id) {
        return  $http.get('/brand/findById?id='+id);
    }
    //增加数据
    this.addBrand=function (brand) {
        return $http.post('/brand/addBrand',brand);
    }
    //修改数据
    this.updateBrand=function (brand) {
        return $http.post('/brand/updateBrand',brand);
    }
    //删除数据
    this.deleteBrand=function (selectIds) {
        return $http.get('/brand/deleteBrand?ids='+selectIds);
    }
    //条件查询
    this.search=function (page,rows,searchEntity) {
        return $http.post('/brand/search?page='+page+"&rows="+rows,searchEntity);
    }
    //下拉列表
    this.selectOptionList=function(){
        return $http.get('/brand/selectOptionList');
    }

});